cdap-gateway/src/main/java/co/cask/cdap/gateway/discovery/UserServiceEndpointStrategy.java

```diff
@@ -80,10 +80,13 @@ public Discoverable pick() {
     String minMaxVersion = null;
     Iterator<Discoverable> iterator = serviceDiscovered.iterator();
     Discoverable result = null;
-    double resultProbability = 0;
+    int wSum = 0;
     while (iterator.hasNext()) {
       Discoverable candidate = iterator.next();
       String version = Bytes.toString(candidate.getPayload());
+      int weight = fetchWeight(version, routeConfig);
```

Comment:
These three lines should be after the `if` block to avoid unnecessary calculation